### PR TITLE
check csi insatll

### PR DESCRIPTION
**Pull request by xinyanw409**
Check whether vsphere csi driver is installed when installing velero plug-in for vsphere. Give warning message if not.

### CODE REVIEW

**Comment by lintongj** [comment]
A few of general comments here.
1. Just confirmed with CSI folks and the vsphere-csi-controller will eventually moved from statefulset to deployment. So, we need be careful about the forward compatibility. For example, we can check either statefulset or deployment is ready or not.
2. We also need to check the version of CSI driver (v1.0.2 and above)
3. It is better to make the check before installing data manager daemenset.

File: pkg/cmd/cli/install/install.go [hunk 1]
```diff
@@ -130,6 +131,56 @@ func NewCommand(f client.Factory) *cobra.Command {
...
+	}
+	statefulsetList, err := clientset.AppsV1().StatefulSets("kube-system").List(metav1.ListOptions{})
+	if err != nil {
+		return false, false, err
+	}
+	for _, item := range statefulsetList.Items {
+		if item.GetName() == "vsphere-csi-controller" {
+			isVersionOk := false
+			for _, container := range item.Spec.Template.Spec.Containers {
+				if strings.Contains(container.Image, "gcr.io/cloud-provider-vsphere/csi/release/driver") {
```
Comment (lintongj) [inline pkg/cmd/cli/install/install.go]: gcr.io/cloud-provider-vsphere/csi/release/driver:v1.0.2
gcr.io/cloud-provider-vsphere/csi/release/syncer:v1.0.2

Make sure both images are v1.0.2 or above

File: pkg/cmd/cli/install/install.go [hunk 2]
```diff
@@ -130,6 +132,54 @@ func NewCommand(f client.Factory) *cobra.Command {
 	return c
 }
 
+func CheckVersion(image string) bool {
```
Comment (lintongj) [inline pkg/cmd/cli/install/install.go]: The version check will return false if the version is v2.0.0. That's not right. So, please refer to Internet for some idea about version check. I believe there is a bunch utilities available.

BTW, the next version of CSI controller is just v2.0.0

Comment (lintongj) [inline pkg/cmd/cli/install/install.go]: The version check will return false if the version is v2.0.0 or v2.0.1. That's not right. So, please refer to Internet for some idea about version check. I believe there is a bunch utilities available.

BTW, the next version of CSI controller is just v2.0.0

Comment (xinyanw409) [reply pkg/cmd/cli/install/install.go]: For vsphere-csi-driver, version prior to v1.0.2: v0.2.0, v0.2.1, v1.0.1, so here just check for these specific cases for simplicity.

File: pkg/utils/constants.go [hunk 3]
```diff
@@ -66,4 +66,9 @@ const (
 // configuration constants for the S3 repository
 const (
 	DefaultS3RepoPrefix = "backups/vsphere-volumes-repo"
+)
+
+// Minimum csi driver version number to meet velero plugin requirement
+const (
+	CSI_MIN_VERSION = "v1.0.2"
```
Comment (dsu-igeek) [inline pkg/utils/constants.go:73]: CamelCase please

File: pkg/cmd/cli/install/install.go [hunk 4]
```diff
@@ -130,6 +133,82 @@ func NewCommand(f client.Factory) *cobra.Command {
...
+	} else {
+		return tag
+	}
+}
+
+// If currentVersion == minVersion, return 0
+// If currentVersion > minVersion, return 1
+// If currentVersion < minVersion, return 2
+// Assume input string format is vX.Y.Z
+func CompareVersion(currentVersion string, minVersion string) int {
```
Comment (dsu-igeek) [inline pkg/cmd/cli/install/install.go:161]: This is the same as the func in the other PR, right?  This should return -1.  We'll merge the velero version check first.  After we merge the velero version check, rebase this PR

Comment (xinyanw409) [reply pkg/cmd/cli/install/install.go:161]: Yes there are two func the same as the func in velero version check pr. I'll rebase and resolve conflict after velero version check pr is merged.

**Pull request merged**